packages/sdk/src/mb-impl/market.ts

```diff
@@ -0,0 +1,67 @@
+// Mintbase marketplace contract JS implementation
+
+import { utils } from 'near-api-js';
+import { TransactionArgs, TransactionAttachments } from '../calls';
+import {
+  DEPOSIT_CONSTANTS,
+  GAS_CONSTANTS,
+  MARKET_METHOD_NAMES,
+} from './constants';
+import { BuyArgs, DepositStorageArgs, ListArgs } from './market.types';
+
+export const buy = (
+  args: BuyArgs,
+): TransactionArgs & TransactionAttachments => {
+  const { nftContractId, tokenId, referrerId, marketAddress, price } = args;
+  return {
+    contractAddress: marketAddress,
+    args: {
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      nft_contract_id: nftContractId,
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      token_id: tokenId,
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      referrer_id: referrerId,
+    },
+    methodName: MARKET_METHOD_NAMES.BUY,
+    gas: GAS_CONSTANTS.MAX_GAS,
+    deposit: price.amount,
```

Comment:
Only applies for non-FT buy

Reply:
Right, can you point me out the right interface for FT buys?

Reply:
This will be done using `ft_transfer_call` on the FT contract: https://github.com/Mintbase/interop-market/blob/main/tests/__tests__/ft-offers.ava.ts#L253